src/google/protobuf/compiler/cpp/helpers.cc

```diff
@@ -1116,7 +1116,7 @@ bool IsAnyMessage(const Descriptor* descriptor, const Options& options) {
 }
 
 bool IsWellKnownMessage(const FileDescriptor* file) {
-  static const std::unordered_set<std::string> well_known_files{
+  static const absl::flat_hash_set<std::string> well_known_files{
```

Comment:
global with destructor is also not safe.  Can you file a bug for both of these (preexisting) issues?

Reply:
Fixed :)